core-metrics/src/main/java/org/apache/kylin/metrics/QuerySparkMetrics.java

```diff
@@ -0,0 +1,398 @@
+                .maximumSize(1000)
+                .expireAfterWrite(600, TimeUnit.SECONDS)
+                .removalListener(new RemovalListener<String, QueryExecutionMetrics>() {
+                    @Override
+                    public void onRemoval(RemovalNotification<String, QueryExecutionMetrics> notification) {
+                        logger.debug("Query metrics {} is removed due to {}",
+                                notification.getKey(), notification.getCause());
+                    }
+                }).build();
+
+    }
+
+    public static QuerySparkMetrics getInstance() {
+        return instance;
+    }
+
+    public void onJobStart(String queryId, String sparderName, long executionId,
+                           long executionStartTime, int jobId, long jobStartTime) {
+        QueryExecutionMetrics queryExecutionMetrics = queryExecutionMetricsMap.asMap().get(queryId);
+        if (queryExecutionMetrics == null) {
+            queryExecutionMetrics = new QueryExecutionMetrics();
+            ConcurrentMap<Integer, SparkJobMetrics> sparkJobMetricsMap = Maps.newConcurrentMap();
+            queryExecutionMetrics.setQueryId(queryId);
+            queryExecutionMetrics.setSparderName(sparderName);
+            queryExecutionMetrics.setExecutionId(executionId);
+            queryExecutionMetrics.setStartTime(executionStartTime);
+            queryExecutionMetrics.setSparkJobMetricsMap(sparkJobMetricsMap);
+            queryExecutionMetricsMap.put(queryId, queryExecutionMetrics);
+        }
+        SparkJobMetrics sparkJobMetrics = new SparkJobMetrics();
+        sparkJobMetrics.setExecutionId(executionId);
+        sparkJobMetrics.setJobId(jobId);
+        sparkJobMetrics.setStartTime(jobStartTime);
+
+        ConcurrentMap<Integer, SparkStageMetrics> sparkStageMetricsMap = Maps.newConcurrentMap();
+        sparkJobMetrics.setSparkStageMetricsMap(sparkStageMetricsMap);
+
+        queryExecutionMetrics.getSparkJobMetricsMap().put(jobId, sparkJobMetrics);
+
+    }
+
+    public void onSparkStageStart(String queryId, int jobId, int stageId, String stageType, long startTime) {
+        SparkStageMetrics sparkStageMetrics = new SparkStageMetrics();
+        sparkStageMetrics.setStageId(stageId);
+        sparkStageMetrics.setStageType(stageType);
+        sparkStageMetrics.setStartTime(startTime);
+        queryExecutionMetricsMap.asMap().get(queryId).getSparkJobMetricsMap().get(jobId).getSparkStageMetricsMap().put(stageId, sparkStageMetrics);
+
+    }
+
+    public void updateSparkStageMetrics(String queryId, int jobId, int stageId, boolean isSuccess, long resultSize,
+                                     long executorDeserializeTime, long executorDeserializeCpuTime, long executorRunTime,
+                                     long executorCpuTime, long jvmGCTime, long resultSerializationTime, long memoryBytesSpilled,
+                                     long diskBytesSpilled, long peakExecutionMemory) {
```

Comment:
There are too many parameters, it's more than 12, consider to use a class to indicate.